File: src/2108.find-first-palindromic-string-in-the-array.2231/solution.ts
function firstPalindrome (words: string[]): string {
    return words.find(isPalindrome) || '';
}

function isPalindrome (string:string) {
    let left = 0;
    let right = string.length - 1;
    while (left < right) {
        if (string[left++] !== string[right--]) {
            return false;
        }
    }
    return true;
}

File: src/2108.find-first-palindromic-string-in-the-array.2231/solution.java
class Solution {
    public String firstPalindrome(String[] words) {
        for (String s : words) {
            if (isPalindrome(s)) {
                return s;
            }
        }
        return "";
    }

    private boolean isPalindrome(String s) {
        int left = 0;
        int right = s.length() - 1;
        while (left < right) {
            if (s.charAt(left++) != s.charAt(right--)) {
                return false;
            }
        }
        return true;
    }
}